solarforecastarbiter/reference_forecasts/models.py

```diff
@@ -0,0 +1,189 @@
+"""
+Default processing steps for NOAA weather models.
+
+All functions in this module have the same signature.
+
+The functions accept:
+
+  * latitude : float
+  * longitude : float
+  * elevation : float
+  * init_time : pd.Timestamp
+      Full datetime of a model initialization
+  * start : pd.Timestamp
+  * end : pd.Timestamp
+
+The functions return a tuple of:
+
+  * ghi : pd.Series
+  * dni : pd.Series
+  * dhi : pd.Series
+  * temp_air : pd.Series
+  * wind_speed : pd.Series
+  * resampler : function
+      A function that resamples data to the appropriate frequency.
+      This function is applied to all variables after power is
+      calculated.
+  * solar_position_calculator : function
+      A function that returns solar position at the ghi forecast times
+      (after interpolation, before resampling). The function will return
+      results immediately if solar position is already known or will
+      call a solar position calculation algorithm and then return.
+
+Most of the functions return forecast data interpolated to 5 minute
+frequency. Interpolation to 5 minutes reduces the errors associated with
+solar position.
+"""
+
+from functools import partial
+
+from solarforecastarbiter import pvmodel
+from solarforecastarbiter.io.nwp import load_forecast
+from solarforecastarbiter.reference_forecasts import forecast
+
+
+def hrrr_subhourly_to_subhourly_instantaneous(latitude, longitude, elevation,
```

Comment:
This function pattern works for external reference forecasts that dictate the timestep and interval.  But in that case, why not simply read the external forecast, then have one function that accepts timestep and interval to resample?  It seems awkward to write a new function for each timestep.